localization/ndt_scan_matcher/include/ndt_scan_matcher/pose_array_interpolator.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2015-2019 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NDT_SCAN_MATCHER__POSE_ARRAY_INTERPOLATOR_HPP_
+#define NDT_SCAN_MATCHER__POSE_ARRAY_INTERPOLATOR_HPP_
+
+#include "ndt_scan_matcher/util_func.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+
+#include <deque>
+
+class PoseArrayInterpolator
+{
+private:
+  using PoseWithCovarianceStamped = geometry_msgs::msg::PoseWithCovarianceStamped;
+
+public:
+  PoseArrayInterpolator(
+    rclcpp::Node * node, const rclcpp::Time target_ros_time,
+    const std::deque<PoseWithCovarianceStamped::ConstSharedPtr> & pose_msg_ptr_array,
+    const double & pose_timeout_sec, const double & pose_distance_tolerance_meters);
+
+  PoseArrayInterpolator(
+    rclcpp::Node * node, const rclcpp::Time target_ros_time,
+    const std::deque<PoseWithCovarianceStamped::ConstSharedPtr> & pose_msg_ptr_array);
+
+  void interpolate(
```

Comment:
It is better to crate as a private function.

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1757/commits/b255b43cde6c73522ef6cd4a66bd2af58cdffedb